federationtypes.go

```diff
@@ -340,6 +404,12 @@ type respSendJoinFields struct {
 
 // ToRespState returns a new RespState with the same data from the given RespSendJoin
 func (r RespSendJoin) ToRespState() RespState {
+	if len(r.StateEvents) == 0 {
+		r.StateEvents = []Event{}
+	}
+	if len(r.AuthEvents) == 0 {
+		r.AuthEvents = []Event{}
```

Comment:
Reading this all the time is getting old. It would probably be better to have a `NewWhateverStruct(...)` which `make(...)`s them to guarantee non-nilness. When unmarshalling, I'd make a custom `UnmarshalJSON` at this point. Ultimately, you only care about this at the JSON boundary, so try to keep this cruft scoped to the JSON boundary.

It's too error prone to rely on the current method.